### PR TITLE
Do not lock relation when getting indexes

### PR DESCRIPTION
**Pull request by xuebinsu**

### CODE REVIEW

File: diskquota_utility.c [hunk 1]
```diff
@@ -1163,29 +1164,25 @@ get_rel_oid_list(void)
...
-			indexIds = RelationGetIndexList(relation);
-			if (indexIds != NIL )
+			if (relkind == RELKIND_INDEX)
 			{
-				foreach(l, indexIds)
-				{
-					oidlist = lappend_oid(oidlist, lfirst_oid(l));
+				HeapTuple index_tup = SearchSysCacheCopy1(INDEXRELID, ObjectIdGetDatum(oid));
+				if (HeapTupleIsValid(index_tup)) {
+					Oid table_oid = DatumGetObjectId(SysCacheGetAttr(INDEXRELID, index_tup, Anum_pg_index_indrelid, &isnull));
```
Comment (higuoxing) [inline diskquota_utility.c:1178]: Nit: Coding style.
```suggestion
				if (HeapTupleIsValid(index_tup))
				{
					Oid table_oid = DatumGetObjectId(SysCacheGetAttr(INDEXRELID, index_tup, Anum_pg_index_indrelid, &isnull));
```

File: diskquota_utility.c [hunk 2]
```diff
@@ -1163,29 +1164,25 @@ get_rel_oid_list(void)
...
+					HeapTuple table_tup = SearchSysCacheCopy1(RELOID, table_oid);
+					char table_kind = ((Form_pg_class) GETSTRUCT(table_tup))->relkind;
+					if (table_kind != RELKIND_RELATION || table_kind != RELKIND_MATVIEW)
```
Comment (higuoxing) [inline diskquota_utility.c:1181]: I think it should be `&&`.

```suggestion
					if (table_kind != RELKIND_RELATION && table_kind != RELKIND_MATVIEW)
```

File: diskquota_utility.c [hunk 3]
```diff
@@ -1149,9 +1150,9 @@ get_rel_oid_list(void)
 
 	initStringInfo(&buf);
 	appendStringInfo(&buf,
-			"select oid "
-			" from pg_class"
-			" where oid >= %u and (relkind='r' or relkind='m')",
+			"select oid, relkind "
+			"from pg_class "
+			"where oid >= %u and (relkind='r' or relkind='m' or (relinkind = 'i'))",
```
Comment (higuoxing) [inline diskquota_utility.c:1155]: Nit: Keep coding style cosistent.
```suggestion
			"where oid >= %u and (relkind='r' or relkind='m' or relinkind = 'i')",
```
OR
```suggestion
			"where oid >= %u and ((relkind='r') or (relkind='m') or (relinkind = 'i'))",
```

**Comment by xuebinsu** [comment]
Status: can be merged after solving the comments and rebasing.

**Comment by xuebinsu** [comment]
Closed in favor or PR #100 .

**Pull request closed without merging**